src/App.tsx

```diff
@@ -1,29 +1,25 @@
-import { Component } from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import { FindMovie } from './components/FindMovie';
 
-interface State {
-  movies: Movie[];
-}
+export const App: React.FC = () => {
+  const [movies, setMovies] = useState<Movie[]>([]);
 
-export class App extends Component<{}, State> {
-  state: State = {
-    movies: [],
+  const handleAddMovie = (movieFromServer: Movie) => {
+    if (!movies.some(({ imdbID }) => movieFromServer.imdbID === imdbID)) {
```

Comment:
For such condition, better to create a variable, for example: `isMovieAlreadyAdded`